lib/mongodb.js

```diff
@@ -483,6 +483,32 @@ MongoDB.prototype.updateOrCreate = function updateOrCreate(model, data, options,
 };
 
 /**
+ * Replace if the model instance exists with the same id or create a new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [callback] The callback function
+ */
+MongoDB.prototype.replaceOrCreate = function replaceOrCreate(model, data, options, callback) {
+  var self = this;
+  if (self.debug) {
+    debug('replaceOrCreate', model, data);
+  }
+  var id = self.getIdValue(model, data);
+  var idName = self.idName(model);
+  var oid = self.coerceId(model, id);
+  delete data[idName];
+  this.execute(model, 'update', {_id: oid}, data, {upsert: true}, function (err, info) {
+    if (err)
+      return callback && callback(err);
+    if (self.debug)
+      debug('replaceOrCreate.callback', model, {_id: oid}, data, err, info);
```

Comment:
`err` is always falsy here. I think it's better to move logging on L504-505 above the error check on L502-503?